### PR TITLE
[18.0][MIG] product_harmonized_system: Migration to 18.0

### PR DESCRIPTION
**Pull request by LucasTran380381**

### CODE REVIEW

File: product_harmonized_system/views/hs_code.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        <field name="res_model">product.category</field>
+        <field name="view_mode">list,form</field>
+        <field name="domain">[('hs_code_id', '=', active_id)]</field>
+    </record>
+    <record id="product_template_hs_code_action" model="ir.actions.act_window">
+        <field name="name">Products</field>
+        <field name="res_model">product.template</field>
+        <field name="view_mode">kanban,list,form</field>
+        <field name="domain">[('hs_code_id', '=', active_id)]</field>
+    </record>
```
Comment (henrybackman) [inline product_harmonized_system/views/hs_code.xml]:
```suggestion
    <record id="product_categ_hs_code_action" model="ir.actions.act_window">
        <field name="name">Product Categories</field>
        <field name="res_model">product.category</field>
        <field name="view_mode">list,form</field>
        <field name="domain">[('hs_code_id', '=', id)]</field>
    </record>
    <record id="product_template_hs_code_action" model="ir.actions.act_window">
        <field name="name">Products</field>
        <field name="res_model">product.template</field>
        <field name="view_mode">kanban,list,form</field>
        <field name="domain">[('hs_code_id', '=', id)]</field>
    </record>
```
`active_id` should be replaced with `id`
- https://github.com/odoo/odoo/commit/daf05d48ac76d500aa1285184bdddc4c67641d58

**Review by henrybackman — CHANGES_REQUESTED** [review]

**Review by henrybackman — APPROVED** [review]

File: product_harmonized_system/models/product_product.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def get_hs_code_recursively(self):
+        res = self.env["hs.code"]
+        if self:
```
Comment (TDu) [inline product_harmonized_system/models/product_product.py:15]: Could we really have `self` not set ?
Maybe it is a good time to add some quick unit tests for this module although there is not much code in it.

Comment (LucasTran380381) [reply product_harmonized_system/models/product_product.py:15]: [intrastat_product/models/account_move.py#L155](https://github.com/OCA/intrastat-extrastat/blob/17.0/intrastat_product/models/account_move.py#L155)  
for example:
account.move.line: product_id is not required. so, `rec.product_id` (`self`) could be `null`

**Review by TDu — APPROVED** [review]

File: product_harmonized_system/security/product_hs_security.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="1">
+    <record id="hs_code_company_rule" model="ir.rule">
+        <field name="name">HS Code Company rule</field>
+        <field name="model_id" ref="model_hs_code" />
+        <field
+            name="domain_force"
+        >['|', ('company_id', '=', False), ('company_id', 'in', company_ids)]</field>
```
Comment (StefanRijnhart) [inline product_harmonized_system/security/product_hs_security.xml]:
```suggestion
        >[('company_id', 'in', company_ids + [False])]</field>
```

Comment (LucasTran380381) [reply product_harmonized_system/security/product_hs_security.xml]: Thanks for suggestion, Fixed

**Review by StefanRijnhart — APPROVED** [review]
Thanks!

**Pull request merged**